include/zephyr/init.h

```diff
@@ -92,20 +87,40 @@ struct init_entry {
  *
  * @param init_id Init entry unique identifier.
  * @param init_fn Init function.
- * @param device Device instance (optional).
+ * @param data_ User data (optional).
  * @param level Initialization level.
  * @param prio Initialization priority within @p level.
  *
  * @see SYS_INIT()
  */
-#define Z_INIT_ENTRY_DEFINE(init_id, init_fn, device, level, prio)             \
+#define Z_INIT_ENTRY_DEFINE(init_id, init_fn, data_, level, prio)              \
 	static const Z_DECL_ALIGN(struct init_entry)                           \
 		Z_INIT_ENTRY_SECTION(level, prio) __used __noasan              \
 		Z_INIT_ENTRY_NAME(init_id) = {                                 \
 			.init = (init_fn),                                     \
-			.dev = (device),                                       \
+			.data = (data_),                                       \
 	}
 
+/**
+ * @brief Obtain wrapper function name for SYS_INIT() and SYS_INIT_NAMED().
+ *
+ * @param name Init entry unique identifier.
+ */
+#define Z_SYS_INIT_WRAPPER_FN_NAME(init_id) _CONCAT(__sys_init_, init_id)
+
+/**
+ * @brief Define wrapper function for SYS_INIT() and SYS_INIT_NAMED().
+ *
+ * @param init_id Init entry unique identifier.
```

Comment:
Should we use name instead of init_id? (since we use name for the param everywhere else)